### PR TITLE
supports tcpping and tcpconn and add error handling of connection errors

### PR DESCRIPTION
**Pull request by chyeh**
**[OWL-611] Implementation of supporting `tcpping` & `tcpconn`**

It supports `tcpping` and `tcpconn` now. Probes are executed in corresponding goroutines now. The code was refactored and the unit tests was added. 

**[OWL-659] Not interrupted by errors on connections**

Fixed the problem of being interrupted by the unsuccessful connection to
both the RPC server `hbs` and the HTTP server `agent`.

### CODE REVIEW

**Comment by chyeh** [comment]
I just found that different probes are using the `FpingInterval` field as their invervals. I'll fix it and re-open this PR

File: measure.go [hunk 1]
```diff
@@ -2,49 +2,163 @@ package main
...
+	statsData := Calc(parsedData, u)
+	jsonParams := Marshal(statsData, u)
+	Push(jsonParams, u.utilName())
 }
 
 func Measure() {
-	go fping()
-	go tcpping()
-	go tcpconn()
+	go func() {
```
Comment (hitripod) [inline measure.go]: Could the agent collect the metrics using only `fping`?

Comment (chyeh) [inline measure.go]: How about adding some fields in the `cfg.json`, which let you enable/disable different types of probes?

Comment (hitripod) [inline measure.go]: It should be controlled by HBS not `cfg.json`, please refer to [open-falcon agent](https://github.com/Cepave/agent/blob/develop/cron/collector.go#L28).

```
    for _, v := range funcs.Mappers {
        go collect(int64(v.Interval), v.Fs)
    }
```

**Pull request merged**